dask/array/core.py

```diff
@@ -2179,7 +2185,7 @@ def from_zarr(url, component=None, storage_options=None, chunks=None, **kwargs):
         mapper = url
         z = zarr.Array(mapper, read_only=True, path=component, **kwargs)
     chunks = chunks if chunks is not None else z.chunks
-    return from_array(z, chunks, name='zarr-%s' % url)
+    return from_array(z,  chunks, name = name)
```

Comment:
It would be useful for the default name of the output array to indicate it was created with `from_zarr`. For example, something like

```python
if name is None:
    name = 'from-zarr-' + tokenize(z, component, storage_options, chunks, **kwargs)
```

Reply:
I'm not sure I agree.  The only reason I can think of for decent however, is that establishing protocol for identifying source in a name string seems undesirable, as opposed to say, creating a higher level object to accumulate such information.

Reply:
> creating a higher level object to accumulate such information

The `name` information here is stored in the dask graph representation of the dask array. Using more descriptive graph key names like `'from-zarr-...'`, while not required, can be useful when inspecting the underlying dask graph (especially when debugging) and is generally used throughout the codebase. For example:

https://github.com/dask/dask/blob/a3cba093ea041a999258e518ab5d5f5c93f611b1/dask/dataframe/io/io.py#L181

https://github.com/dask/dask/blob/a3cba093ea041a999258e518ab5d5f5c93f611b1/dask/array/core.py#L2342

https://github.com/dask/dask/blob/a3cba093ea041a999258e518ab5d5f5c93f611b1/dask/dataframe/io/io.py#L253

Reply:
I follow.  Ok thanks. Change pushed.